src/sentry/static/sentry/app/components/events/errors.tsx

```diff
@@ -0,0 +1,242 @@
+  };
+
+  uniqueErrors = errors => uniqWith(errors, isEqual);
+
+  onReprocessEvent = async () => {
+    const {orgId, project, event} = this.props;
+    const endpoint = `/projects/${orgId}/${project.slug}/events/${event.id}/reprocessing/`;
+
+    addLoadingMessage(t('Reprocessing event\u2026'));
+
+    try {
+      await this.props.api.requestPromise(endpoint, {
+        method: 'PUT',
+      });
+    } catch (e) {
+      clearIndicators();
+      addErrorMessage(
+        t('Failed to start reprocessing. The event is likely too far in the past.')
+      );
+      return;
+    }
+
+    clearIndicators();
+    browserHistory.push(
+      `/organizations/${orgId}/issues/?query=tags[original_event_id]:${event.id}`
+    );
+  };
+
+  onReprocessGroup = async () => {
+    const {orgId, issueId} = this.props;
+    const endpoint = `/organizations/${orgId}/issues/${issueId}/reprocessing/`;
+
+    addLoadingMessage(t('Reprocessing issue\u2026'));
+
+    try {
+      await this.props.api.requestPromise(endpoint, {
+        method: 'PUT',
+      });
+    } catch (e) {
+      clearIndicators();
+      addErrorMessage(
+        t('Failed to start reprocessing. The event is likely too far in the past.')
+      );
+      return;
+    }
+
+    clearIndicators();
+    browserHistory.push(
+      `/organizations/${orgId}/issues/?query=tags[original_group_id]:${issueId}`
+    );
+  };
+
+  onReprocessStart = () => {
+    openModal(this.renderReprocessModal);
+  };
+
+  renderReprocessModal = ({Body, closeModal}) => (
+    <React.Fragment>
+      <Body>
+        <p>
+          {t(
+            'You can choose to re-process events to see if ' +
+              'your errors have been resolved. Keep the following limitations in mind:'
+          )}
+        </p>
+
+        <ul>
+          <li>
+            {t(
+              'Sentry will duplicate events in your project (for now) and not delete the old versions.'
+            )}
+          </li>
+          <li>
+            {t(
+              'Reprocessing one or multiple events counts against your quota, but bypasses rate limits.'
+            )}
+          </li>
+          <li>
+            {t(
+              'If an event is reprocessed but did not change, we will not create the new version and not bill you for it (for now).'
+            )}
+          </li>
+          <li>
+            {t(
+              'If you have provided missing symbols please wait at least 1 hour before attempting to re-process. This is a limitation we will try to get rid of.'
+            )}
+          </li>
+        </ul>
+      </Body>
+      <div className="modal-footer">
+        <Button onClick={this.onReprocessGroup}>
+          {t('Reprocess all events in issue')}
+        </Button>
+        <Button style={{marginLeft: space(1)}} onClick={this.onReprocessEvent}>
+          {t('Reprocess single event')}
+        </Button>
+        <Button style={{marginLeft: space(1)}} onClick={closeModal}>
```

Comment:
I don't see how does this look, but should these buttons be color-coded? E.g. one action be primary, etc..

Reply:
i dont know which action is primary in this case unfortunately. I assume Cancel, as it's the safest?

Reply:
right now buttons are ordered from (most impact) to (least impact)